backstory/src/main/java/com/google/sps/story/StoryManagerImpl.java

```diff
@@ -144,4 +156,22 @@ private String makeRequestBody(String prefix, int maxLength, Double temperature)
     String convertedMap = gson.toJson(requestMap);
     return convertedMap;
   }
+
+  /**
+   * Cycles selectedUrlIndex to a backup url.
+   */
+  public void cycleUrl() {
+    if (selectedUrlIndex < 4) {
```

Comment:
Using the literal 4 here makes this more likely to break if the serviceUrls array is updated. Could you get the size of the array instead?

Reply:
Done